primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -87,16 +87,17 @@ impl<G: Group, W: PedersenWindow, const N: usize> BoweHopwoodPedersenCRH<G, W, N
     }
 
     /// Zcash protocol specification - section 5.4.1.7
-    fn calculate_num_chunks_in_segment<F: PrimeField>() -> usize {
-        let upper_limit: BigUint = F::modulus_minus_one_div_two().into();
+    pub(crate) fn calculate_num_chunks_in_segment<F: PrimeField>() -> usize {
+        let upper_limit: BigUint = ((F::modulus_minus_one_div_two().into() * 15u32)/4u32) + 1u32;
```

Comment:
Not sure that dividing by 4 will not introduce rounding issues; I would check the inequality as follows: `4*2^4c <= F::modulus_minus_one_div_two*15-4 `: this should not introduce rounding as 2^4c-1 should always be divisible by 15, and it requires in each iteration only an additional shift of 2 bits on range when comparing with `upper_limit`